File: composeApp/src/androidMain/kotlin/cz/miroslavpasek/pigeonnavigator/ui/search/screen/SearchContent.kt
package cz.miroslavpasek.pigeonnavigator.ui.search.screen

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.Button
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import cz.miroslavpasek.pigeonnavigator.ui.common.theme.AppSpacing

/**
 * Stateless layout for the standalone search screen.
 *
 * Receives the rendered text only and forwards user inputs through the supplied
 * callbacks; persistence/Store wiring lives in [SearchScreen].
 */
@Composable
fun SearchContent(
    query: String,
    isLoading: Boolean,
    errorMessage: String?,
    results: List<String>,
    onQueryChange: (String) -> Unit,
    onClear: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .fillMaxSize()
            .padding(AppSpacing.md),
        verticalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        Row(horizontalArrangement = Arrangement.spacedBy(AppSpacing.sm)) {
            OutlinedTextField(
                modifier = Modifier.weight(1f),
                value = query,
                onValueChange = onQueryChange,
                label = { Text("Search") },
                singleLine = true,
            )
            Button(onClick = onClear) {
                Text("Clear")
            }
        }

        if (isLoading) {
            CircularProgressIndicator()
        }

        errorMessage?.let { Text(it) }

        LazyColumn(modifier = Modifier.fillMaxWidth()) {
            items(results) { item ->
                Text(
                    text = item,
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(vertical = AppSpacing.sm),
                )
            }
        }
    }
}

File: iosApp/iosApp/UI/Search/Screen/SearchContent.swift
import SwiftUI

/// Stateless layout for the standalone search screen.
///
/// Receives strings/booleans only and forwards user input through callbacks;
/// the owning `SearchScreen` is responsible for store wiring.
struct SearchContent: View {
    @Binding var query: String
    let isLoading: Bool
    let errorMessage: String?
    let results: [String]
    let onQueryChange: (String) -> Void
    let onSubmit: () -> Void
    let onClear: () -> Void

    var body: some View {
        VStack(spacing: 12) {
            HStack(spacing: 8) {
                TextField("Search places", text: $query)
                    .textFieldStyle(.roundedBorder)
                    .onChange(of: query) { newValue in
                        onQueryChange(newValue)
                    }
                    .submitLabel(.search)
                    .onSubmit { onSubmit() }

                Button("Search") { onSubmit() }
                Button("Clear") { onClear() }
            }

            if isLoading {
                ProgressView()
            }

            if let errorMessage = errorMessage {
                Text(errorMessage)
                    .font(.footnote)
                    .foregroundStyle(.red)
                    .frame(maxWidth: .infinity, alignment: .leading)
            }

            List(results, id: \.self) { item in
                Text(item)
            }
            .listStyle(.plain)
        }
        .padding()
    }
}
